server/gunicorn/gunicorn.conf.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+for k,v in os.environ.items():
+    if k.startswith("GUNICORN_"):
+        key = k.split('_', 1)[1].lower()
+        locals()[key] = v
```

Comment:
would you be up for walking me through what this is doing?

Reply:
Yep! So it's exposing gunicorn config options to gunicorn (options are defined [here](https://docs.gunicorn.org/en/stable/run.html)). We defined them as environment variables, which are usually in THIS_CASE. It's picking up the ones prefixed with GUNICORN, lowercasing them as gunicorn expects them, and setting them on the [local symbol table](https://docs.python.org/3/library/functions.html#locals), which apparently is how gunicorn reads them. Tbh, I'm not sure without more research why gunicorn is looking at `locals()`.

Reply:
I think it's because there's no direct way to provide these args via ENV vars, which is the only way for people using the BMX image this come from can use. So, it's sort of inject them as if they were read in from a config file.